src/TinyRAM/Instructions.hs

```diff
@@ -401,3 +374,30 @@ readInputTape ri a = do
       setRegisterValue ri 0
       setConditionFlag 1
   incrementProgramCounter
+
+
+alignToWord :: WordSize -> Address -> (Address, Integer)
+alignToWord ws address =
+  (address - fromIntegral offset, toInteger offset)
+  where
+    offset = fromIntegral address `rem` bytesPerWord ws
+
+
+setByte :: Word -> Integer -> Integer -> Word
+setByte word offset val =
+  (word .&. mask) .|. val'
+  where
+    mask :: Word
+    mask = complement (0xff `shift` shift')
+
+    val' :: Word
```

Comment:
Why is `val'` computed this way? Should it not be the result of bitwise anding `val` together with the complement of `mask`? I feel like this computation of `val'` is only correct when setting the first byte, and maybe this is wrong and also points to a gap in the test suite.